File: 2904-shortest-and-lexicographically-smallest-beautiful-string/2904-shortest-and-lexicographically-smallest-beautiful-string.java
class Solution {
    public String shortestBeautifulSubstring(String s, int k) {
        List<Integer> pos = new ArrayList<>();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '1') {
                pos.add(i);
            }
        }
        if (pos.size() < k)
            return "";
        String ans = "";
        for (int i = 0; i + k - 1 < pos.size(); i++) {
            int start = pos.get(i);
            int end = pos.get(i + k - 1);
            String cur = s.substring(start, end + 1);
            if (ans.isEmpty() ||
                cur.length() < ans.length() ||
                (cur.length() == ans.length() && cur.compareTo(ans) < 0)) {
                ans = cur;
            }
        }
        return ans;
    }
}

File: 2904-shortest-and-lexicographically-smallest-beautiful-string/2904-shortest-and-lexicographically-smallest-beautiful-string.cpp
class Solution {
public:
    string shortestBeautifulSubstring(string s, int k) {
        vector<int> pos;
        for (int i = 0; i < s.size(); i++) {
            if (s[i] == '1') {
                pos.push_back(i);
            }
        }
        if (pos.size() < k)
            return "";

        string ans = "";
        for (int i = 0; i + k - 1 < pos.size(); i++) {
            int start = pos[i];
            int end = pos[i + k - 1];
            string cur = s.substr(start, end - start + 1);
            if (ans == "" ||
                cur.size() < ans.size() ||
                (cur.size() == ans.size() && cur < ans)) {

                ans = cur;
            }
        }
        return ans;
    }
};